src/util/modp_numtoa.c

```diff
@@ -141,6 +141,13 @@ void modp_ftoa(float value, char* str, int prec)
                 prec = 9;
         }
 
+	/* check for nan */
+	if ( value != value )
```

Comment:
So, this is going to blow up things that expect an actual number.  I believe we use this function to populate values in JSON.  With this change, we'll have something like this, which is invalid JSON:

{
	"x": nan
}

Does the original behavior output zero? Any ideas what it could output that's closer to a real value?

Reply:
Yes.  The original behaviour was to print "0.0".  I see the problem here.  Python throws an exception, whereas C happily runs code that divides by zero.  Technically this is not needed for the -Wdouble_promote.  It would be nice to see bad values when debugging, so probably safer to omit this although it would be handy to catch cases where we divide by 0 in firmware.  #ifdef DEBUG?

Reply:
Retaining the original behavior and maybe doing an assert if in debug mode could be a good approach.  Check to see if we're already doing asserts elsewhere in the firmware, I think we are...

Reply:
I don't see any assert usage.   Alternatively, it looks like the pr_..._float macros are the only ones that use printk_float_msg.  I could catch my bad floats there instead.